apps/frontend/src/modules/workspace/hooks/useWorkspace.tsx

```diff
@@ -58,14 +60,54 @@ export const useWorkspace = () => {
                         },
                     },
                 });
-                return response?.data?.organizationUpdate;
+                if (response?.data?.organizationUpdate) {
+                    const updatedUser = JSON.parse(JSON.stringify(user)) as DeepPartial<User>;
+                    const currentOrganization = updatedUser.organizations?.edges?.find(
+                        (org) => org?.node?.id === workspace?.id,
+                    );
+                    if (currentOrganization) {
+                        currentOrganization.node = {
+                            ...currentOrganization.node,
+                            name: organizationInput.name || "",
```

Comment:
@afolasope this way, if `organizationInput.name` or `organizationInput.slug` is `undefined` it sets it to an empty string which is not what we want. If the name or slug wasn't updated, it should remain the previous value.